content/200-concepts/100-components/02-prisma-client/051-working-with-fields/working-with-json-fields.mdx

```diff
@@ -46,134 +44,746 @@ export declare type JsonValue = string | number | boolean | null | JsonObject |
+
+    if (i && typeof i === 'object' && Array.isArray(i)) {
+      const insurancesArray = i as Prisma.JsonArray
+
+      insurancesArray.forEach(i => {
+        if (i && typeof i === 'object' && !Array.isArray(i)) {
+          const insuranceObject = i as Prisma.JsonObject
+
+          insuranceObject['status'] = 'expired'
+        }
+      })
+
+      const whereClause = Prisma.validator<Prisma.UserWhereInput>()({
+        id: user.id,
+      })
+
+      const dataClause = Prisma.validator<Prisma.UserUpdateInput>()({
+        extendedPetsData: petsObject,
+      })
+
+      userQueries.push(
+        prisma.user.update({
+          where: whereClause,
+          data: dataClause,
+        })
+      )
+    }
+  }
+})
+
+if (userQueries.length > 0) {
+  console.log(userQueries.length + ' queries to run!')
+  await prisma.$transaction(userQueries)
+}
+
+const json = [{ status: 'expired', insuranceID: 92 }]
+
+const checkJson = await prisma.user.findMany({
+  where: {
+    extendedPetsData: {
+      path: ['insurances'],
+      array_contains: json,
     },
   },
 })
-```
\ No newline at end of file
+
+console.log(checkJson.length)
+```
+
+</tab>
+
+<tab>
+
+```ts
+const userQueries: string | any[] = []
+
+getUsers.forEach(user => {
+  if (
+    user.extendedPetsData &&
+    typeof user.extendedPetsData === 'object' &&
+    !Array.isArray(user.extendedPetsData)
+  ) {
+    const petsObject = user.extendedPetsData as Prisma.JsonObject
+
+    const i = petsObject['insurances']
+
+    if (i && typeof i === 'object' && Array.isArray(i)) {
+      const insurancesArray = i as Prisma.JsonArray
+
+      insurancesArray.forEach(i => {
+        if (i && typeof i === 'object' && !Array.isArray(i)) {
+          const insuranceObject = i as Prisma.JsonObject
+
+          insuranceObject['status'] = 'expired'
+        }
+      })
+
+      const whereClause = Prisma.validator<Prisma.UserWhereInput>()({
+        id: user.id,
+      })
+
+      const dataClause = Prisma.validator<Prisma.UserUpdateInput>()({
+        extendedPetsData: petsObject,
+      })
+
+      userQueries.push(
+        prisma.user.update({
+          where: whereClause,
+          data: dataClause,
+        })
+      )
+    }
+  }
+})
+
+if (userQueries.length > 0) {
+  console.log(userQueries.length + ' queries to run!')
+  await prisma.$transaction(userQueries)
+}
+
+const json = { status: 'expired', insuranceID: 92 }
+
+const checkJson = await prisma.user.findMany({
+  where: {
+    extendedPetsData: {
+      path: '$.insurances',
+      array_contains: json,
+    },
+  },
+})
+
+console.log(checkJson.length)
+```
+
+</tab>
+</TabbedContent>
+
+## <inlinecode>Json</inlinecode> FAQs
+
+### Can you select a subset of JSON key/values to return?
+
+No - it is not yet possible to select which JSON elements to return. Prisma Client returns the entire JSON object.
+
+### Can you filter on the presence of a specific key?
+
+No - it is not yet possible to filter on the presence of a specific key.
```

Comment:
if we say "not yet", can we link to github issues that request this so that people can subscribe when they're interested in this feature? (if we don't have gh issues, would it be worth creating them ourselves?)